### DISCUSSION TITLE
[Feature] Social Sharing from all views in mobile app

### DISCUSSION THREAD

**Discussion by ja2ui0** [immich-app/immich#2346] [Feature Request]
### The feature

As an Immich user trying to increase my usage of the app to manage photos in most scenarios, I would like the mobile app to allow "sharing" in the typical way to various destination, such as Pixelfed, Telegram, and other social media or communication platforms.

In the web views, it's easy enough to download and attach a photo or copy it to the clipboard and paste it, but on android, one must resort to the phone's native albums to share content - re-downloading photos if necessary to accomplish it. The ability to do this natively in Immich would effectively replace most daily use cases for Android's albums for me.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by Morethanevil** [comment]
Immich has a system sharemenu. Click on a Picture in the mobile app. On left side is the Button

**Reply by ja2ui0** [reply]
Whoops, my bad. The icon is not what I expected. I'll close this.

**Reply by jon-f-novabackup** [reply]
I mean to be fair the mobile icon to share an image is not the greatest icon design to use to be recognized universally (like a "network" style icon would), it looks very close to the upload and the archive icons in fact, and there is no text description for any of those action icons at the bottom. It should in my opinion look more like a "networking" style of icon like this gphotos share icon at the bottom left (but without the text in this case) and below that screenshot is the Share icon used by GitHub mobile:

![image](https://user-images.githubusercontent.com/50121594/235217777-13ee2360-f09e-42f7-9558-32dae5a42f51.png)

And this is the Share icon used by GitHub mobile:

![Screenshot_20230429-080947.png](https://user-images.githubusercontent.com/50121594/235310468-ab4cd028-5534-4506-9490-e854a26512d0.png)

The current Immich 1.54.0 build.77 mobile app "Share" asset button icon looks very similar to the upload button on mobile and a bit like the archive button on mobile, here is the current share icon in Immich; please note that in 1.54.0 mobile there is no text description under the "Share" icon in the Albums (Library) and Sharing views, the text for the Share icon is only visible in the Photos view while viewing the thumbnails and selecting one ore more assets, then it does show the Share icon with the word "Sharing" under it:

![Screenshot_20230428-085959~2.png](https://user-images.githubusercontent.com/50121594/235197210-60c1c56b-41ce-4678-b773-ad396c58ca63.png)

Note: I feel like I'm guessing what certain icons do very often in Immich mobile, and that is partly due to having no text that describes each icon, but mainly it is just due to the look of the icon in some cases that is not obvious, like the Share button icon.  An alternative to replacing the Sharing icon with another better / easier to recognize that it equates to a "Share" function, would be to just add text descriptions under each icon universally so it is the same across all menus in mobile. The Photos view in mobile has the button described for the "Share" button function, but that did not carry over to the other views in Albums and Sharing views.

**Comment by ja2ui0** [comment]
That's exactly what I was expecting it to look like!

**Reply by jon-f-novabackup** [reply]
I created a feature request for this one just now as https://github.com/immich-app/immich/discussions/2348. Note: It would also be nice to in the Albums thumbnails view and Sharing thumbnails view to be able to click to select each image that you want to share and then click once on a Share button (which only exists in Photos view now in 1.54.0 build.77), so that you don't have to view each asset to individually share each asset, one by one in Albums / Shared thumbnails view. The Photos thumbnails view already has this capability in the mobile app, and if you do utilize that method of Photos view thumbnail select it will automatically show a Share icon below that **you will notice that the Share icon in Photos view does have a text description under each button icon** to realize what the Share and other buttons do; the Share icon there is the same as discussed in the Albums and Sharing views in mobile however, just without a description text added in the later two views. The buttons in the other views such as the Share button icon when you are viewing individual images / assets discussed here do not have text descriptions of the buttons showing in the bottom menu, only in Photos view does a text description for the icons for buttons exist.

**Comment by ja2ui0** [comment]
That's perfect, thanks for grooming this.
